### ISSUE TITLE
FilterProps is missing in the glamorous.d.ts file

### ISSUE THREAD

**Issue by johnjesse** [paypal/glamorous#339]
- `glamorous` version: 4.9.7
- `glamor` version: 2.20.40
- `react` version: 15.6
- `typescript` version: 2.5.3

Problem description:

I'm trying to style a component with a prop, and stop this being forwarded onto the component

```typescript
const MyStyledComponent = glamorous(MyComponent, {filterProps: ['extraPropForStyling'] })<IMyComponentProps & IExtraStylingProps>(({ extraPropForStyling }) => {
    // Do styling stuff here
});
```

This causes a ts compiler error (which is quite unhelpful)
` Argument of type 'GlamorousComponent<Pick<IMyComponentProps , never> & HTMLProps<HTMLSpanElement>, IMyComponentProps>' is not assignable to parameter of type '"tspan"'.`

but it does work, The prop is not getting forwarded.

After some digging I think the error is just that `filterProps` is not included on the `GlamorousOptions` interface.

From `glamorous.d.ts`

```
export interface GlamorousOptions<Props, Context, DefaultProps> {
  displayName: string
  rootEl: string | Element
  forwardProps: String[]
  shouldClassNameUpdate: (
    props: Props,
    prevProps: Props,
    context: Context,
    prevContext: Context,
  ) => boolean
  propsAreCssOverrides?: false
  withProps: DefaultProps
}
```

Suggested solution:

add it to the interface
`filterProps: String[];`

I'd be happy to submit a PR, it'd be my first but gotta start somewhere

**Comment by luke-john** [comment]
Hey @johnjesse, thanks for the detailed issue and the investigation.

Just looking into this and it seems that the typings aren't actually working when you pass the generic after the component (ie `glamorous(Component)<Props>`) rather than before (ie `glamorous<Props>(Component)`. When you pass it before it also ensures the types match the component passed.

So while adding `filterProps: string[];` (it should be lower case, the uppercase on `forwardProps` is a seperate bug) appears to work for the usage your looking for, to get full typesafety we'll need to do something similar to `PropsAreCssOverridesGlamorousOptions`.

I've got a product launch coming up at work, and am unlikely to get much time to investigate that or the other generics issue that your investigation has exposed for the next two weeks.  A PR with your suggestion would be welcome and we can add something to the typescript usage notes about the issues people might have with it.

If someone else wants to look into giving the full type coverage as well that would be excellent.

**Comment by johnjesse** [comment]
Cool, I've submitted a pull request to fix the filterProps, and forwardProps typing issues

**Comment by RoystonS** [comment]
I think this can be closed: PR #340 was merged on 20 October has made it into a release.

**Comment by kentcdodds** [comment]
This project is now in an unmaintained status. Please see the README for more information.
